### PR TITLE
Lambda expressions / it-academy.by / JC1-110-22 / Alena Trafimovich / Done tasks

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelper.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ArgumentValueHelper {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelper.java:10]: Remove blank line

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelper.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return map;
+    }
+
+    public String[] getStringArrayFromMap(Map<String, String> map) {
+        List<String> list = new ArrayList<>();
+        map.forEach((key, value) -> {
+            list.add(key);
+            list.add(value);
+        });
+        return list.toArray(new String[0]);
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelper.java:27]: Use Stream API

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelperTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+public class ArgumentValueHelperTest {
+    private final ArgumentValueHelper argumentValueHelper = new ArgumentValueHelper();
+    private String[] text;
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelperTest.java:13]: move to local var

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelperTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+public class ArgumentValueHelperTest {
+    private final ArgumentValueHelper argumentValueHelper = new ArgumentValueHelper();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/ArgumentValueHelperTest.java:12]: move to `@BeforeEach`

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class CityHelper {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java:9]: move blank line

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MultiplicationTable.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import java.util.stream.IntStream;
+
+public class MultiplicationTable {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/MultiplicationTable.java:6]: move blank line

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    public Set<String> getUniqueCities(List<String> cities) {
+        return new HashSet<>(cities);
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/CityHelper.java:12]: this is dirty hack

File: src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticsTest.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+import java.util.Arrays;
+
+public class MathStatisticsTest {
+    private final MathsStatistics mathsStatistics = new MathsStatistics();
+
+    @Test
+    public void testGetRandomNumbers() {
+        Arrays.stream(MathsStatistics.getRandomNumbers(10)).boxed().forEach(System.out::println);
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/MathStatisticsTest.java:14]: What is it?

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package com.rakovets.course.java.core.practice.lambda_expressions;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TvSeller {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java:8]: remove blank line

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        return tvs.stream()
+                .filter(c -> c.getPrice() >= lowerBound && c.getPrice() <= higherBound)
+                .collect(Collectors.toList());
+    }
+
+    public List<Tv> getSortedByPrice(List<Tv> tvs, boolean isAscending) {
+        Comparator<Tv> byPrice = Comparator.comparing(Tv::getPrice);
+        if (!isAscending) {
+            byPrice = Comparator.comparing(Tv::getPrice).reversed();
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java:37]: use ternary operator

File: src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        return tvs.stream()
+                .sorted(byPrice)
+                .collect(Collectors.toList());
+    }
+
+    public List<Tv> getSortedByDiagonal(List<Tv> tvs, boolean isAscending) {
+        Comparator<Tv> byDiagonal = Comparator.comparing(Tv::getDiagonal);
+        if (!isAscending) {
+            byDiagonal = Comparator.comparing(Tv::getDiagonal).reversed();
+        }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/lambda_expressions/TvSeller.java:47]: use ternary operator

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Comment by AlenaTrafimovich** [comment]
Fixed

**Pull request closed without merging**